simple_graph/test_wgraph.py

```diff
@@ -0,0 +1,172 @@
+    """Test deletesthenode'n' from the graph, raises an error if no such node exists"""
+    populated_graph.del_node('a')
+    assert not populated_graph.has_node('a')
+    assert 'a' not in populated_graph.dict['c']
+
+
+def test_del_edge_empty(empty_graph):
+    """Test deletes the edge connecting non-existing 'n1' and 'n2' from the graph,
+    raises an error if no such edge exists"""
+    with pytest.raises(AttributeError) as message:
+        empty_graph.del_edge('a', 'b')
+    assert 'No Such Edge Exists' in str(message.value)
+
+
+
+def test_del_edge_empty(populated_graph):
+    """Test deletes the edge connecting 'n1' and 'n2' from the graph"""
+    populated_graph.del_edge('a', 'b')
+    assert 'b' not in populated_graph.dict['a']
+
+
+def test_neighbors(populated_graph):
+    """Test returns the list of all nodes connected to 'n' by edges, raises an error if n is not in g"""
+    l = ['a', 'b']
+    for node in l:
+        assert node in populated_graph.neighbors('c')
+
+
+
+def test_adjacent_empty(empty_graph):
+    """Test returns True if there is an edge connecting n1 and n2, False if not, raises an error if
+    either of the supplied nodes are not in g"""
+    with pytest.raises(KeyError) as message:
+        empty_graph.adjacent('a', 'b')
+    assert 'Node(s) not in graph.' in str(message.value)
+
+
+def test_adjacent_pop(populated_graph):
+    """Test returns True if there is an edge connecting n1 and n2, False if not, raises an error if
+    either of the supplied nodes are not in g"""
+    assert populated_graph.adjacent('a', 'b')
+    assert not populated_graph.adjacent('a', 'd')
+
+def test_depth_first_traversal(depth_populated_graph):
+    """Test the depth first tranversal approach for a graph of arbitrary length"""
+    print "depth: " + str(depth_populated_graph.depth_first_traversal('a'))
+    assert depth_populated_graph.depth_first_traversal('a') == ['a', 'b', 'd', 'f', 'e', 'c', 'g']
+
+def test_breadth_first_traversal(breadth_populated_graph):
+    """Test the breadth first tranversal approach for a graph of arbitrary length"""
+    print "breadth: " + str(breadth_populated_graph.breadth_first_traversal('a'))
+    assert breadth_populated_graph.breadth_first_traversal('a') == ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h']
+
+def test_breadth_with_cycle(populated_graph):
+    """Test the depth first tranversal approach for a graph of arbitrary length and a cycle"""
+    assert populated_graph.breadth_first_traversal('a') == ['a', 'b', 'c', 'd']
+
+@pytest.fixture(scope='function')
+def empty_graph():
+    graph = wgraph.Wgraph()
+    return graph
+
+
+@pytest.fixture(scope='function')
+def populated_graph():
+    graph = wgraph.Wgraph()
+    graph.dict = {
+        'a': OrderedDict([('b', 1)]),
+        'b': OrderedDict([('c', 2), ('d', 4)]),
+        'c': OrderedDict([('b', 5), ('a', 3)]),
+        'd': OrderedDict([('c', 6)]),
+        'e': OrderedDict()
+        }
+    return graph
```

Comment:
I like that you are building the graphs in these methods without using the graph api itself.  That ensures that failures in your methods will not effect the structure of what you're expecting to find.